File: PharmaDicBackEnd/PharmaDicBackEnd.ApiService/DTOs/DrugInteractionInputDto.cs
using System.ComponentModel.DataAnnotations;

namespace PharmaDicBackEnd.ApiService.DTOs
{
    public class DrugInteractionInputDto
    {
        [Required(ErrorMessage = "ID Thuốc thứ nhất không được để trống.")]
        public int MedicineId1 { get; set; }

        [Required(ErrorMessage = "ID Thuốc thứ hai không được để trống.")]
        public int MedicineId2 { get; set; }

        [StringLength(50, ErrorMessage = "Mức độ nghiêm trọng không vượt quá 50 ký tự.")]
        public string? Severity { get; set; }

        public string? Description { get; set; }

        public string? Recommendation { get; set; }
    }
}

File: PharmaDicMobile/app/src/main/java/com/example/pharmadicmobile/dtos/DrugInteractionInputDto.java
package com.example.pharmadicmobile.dtos;

public class DrugInteractionInputDto {
    private int medicineId1;
    private int medicineId2;
    private String severity;
    private String description;
    private String recommendation;

    public DrugInteractionInputDto() {}

    public int getMedicineId1() { return medicineId1; }
    public void setMedicineId1(int medicineId1) { this.medicineId1 = medicineId1; }

    public int getMedicineId2() { return medicineId2; }
    public void setMedicineId2(int medicineId2) { this.medicineId2 = medicineId2; }

    public String getSeverity() { return severity; }
    public void setSeverity(String severity) { this.severity = severity; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public String getRecommendation() { return recommendation; }
    public void setRecommendation(String recommendation) { this.recommendation = recommendation; }
}
